src/main/java/io/tackle/pathfinder/services/AssessmentSvc.java

```diff
@@ -118,9 +114,76 @@ private Questionnaire defaultQuestionnaire() {
+                    stakegroup.delete();
+                }
+            });
+            // Add not existing stakeholdergroups included in the current array
+            assessmentDto.getStakeholderGroups().forEach(e -> {
+                log.info("Considering Stakeholdergroup : " + e);
+                if (assessment.stakeholdergroups.stream().noneMatch(o -> o.stakeholdergroupId == e)) {
+                    log.info("Adding Stakeholdergroup : " + e);
+                    AssessmentStakeholdergroup.builder()
+                            .assessment(assessment)
+                            .stakeholdergroupId(e)
+                            .build().persist();
+                }
+            });
+        }
+        if (null != assessmentDto.getStakeholders()) {
+            // Delete existing stakeholders not included in current array
+            assessment.stakeholders.forEach(stake -> {
+                if (!assessmentDto.getStakeholders().contains(stake.stakeholderId)) {
+                    log.info("Deleted stake : " + stake.stakeholderId);
+                    stake.delete();
+                }
+            });
+            // Add not existing stakeholders included in the current array
+            assessmentDto.getStakeholders().forEach(e -> {
+                log.info("Considering Stakeholder : " + e);
+                if (assessment.stakeholders.stream().noneMatch(o -> o.stakeholderId == e)) {
+                    log.info("Adding Stakeholder : " + e);
+                    AssessmentStakeholder.builder()
+                        .assessment(assessment)
+                        .stakeholderId(e)
+                        .build().persist();
+                }
+            });
+        }
+
+        assessmentDto.getQuestionnaire().getCategories().forEach(categ -> {
+            AssessmentCategory category = AssessmentCategory.find("assessment_questionnaire_id=?1 and id=?2", assessment.assessmentQuestionnaire.id, categ.getId()).<AssessmentCategory>firstResultOptional().orElseThrow(BadRequestException::new);
+            if (categ.getComment() != null) {
+              category.comment = categ.getComment();
+                log.info("Setting category comment : " + category.comment);
+            }
+
+            categ.getQuestions().forEach(que -> {
+                AssessmentQuestion question = AssessmentQuestion.find("assessment_category_id=?1 and id=?2", categ.getId(), que.getId()).<AssessmentQuestion>firstResultOptional().orElseThrow(BadRequestException::new);
+
+                que.getOptions().forEach(opt -> {
+                    AssessmentSingleOption option = AssessmentSingleOption.find("assessment_question_id=?1 and id=?2", question.id, opt.getId()).<AssessmentSingleOption>firstResultOptional().orElseThrow(BadRequestException::new);
+                    if (opt.getChecked() != null) {
+                      option.selected = opt.getChecked();
+                        log.info("Setting category comment : " + category.comment);
```

Comment:
this log message should be logging the checked field passed in

Reply:
:+1: